xdsl/dialects/csl.py

```diff
@@ -80,6 +82,77 @@ class ModuleKind(StrEnum):
     PROGRAM = "program"
 
 
+class TaskKind(StrEnum):
+    LOCAL = "local"
+    DATA = "data"
+    CONTROL = "control"
+
+
+class _FuncBase(IRDLOperation):
```

Comment:
This should probably also inherit from `ABC` for completeness sake, just so that it's clear that this is not to be instantiated